nanoFramework.Device.Bluetooth/GenericAttributeProfile/GattLocalCharacteristic.cs

```diff
@@ -0,0 +1,400 @@
+            {
+                // LocalCharacteristic event ?
+                if (descritorId == 0)
+                {
+                    handled = true;
+                    WriteRequested?.Invoke(this, e);
+                }
+                else
+                {
+                    // Descriptor event
+                    GattLocalDescriptor des = FindDescriptor(descritorId);
+                    if (des != null)
+                    {
+                        handled = des.OnWriteRequested(e);
+                    }
+                }
+            }
+
+            if (!handled)
+            {
+                // No event handler, respond with error
+                e.GetRequest().RespondWithProtocolError(GattProtocolError.UnlikelyError);
+            }
+        }
+
+        private int ClientSubscribed(GattSubscribedClient client)
+        {
+            for(int index=0; index<_subscribedClients.Count; index++)
+            {
+                if ( ((GattSubscribedClient)_subscribedClients[index]).Session.DeviceId.Id == client.Session.DeviceId.Id)
+                {
+                    return index;
+                }
+            }
+            return -1;
+        }
+
+        internal void OnSubscribedClientsChanged(bool subscribe, GattSubscribedClient client)
+        {
+            lock (_subscribedClients)
+            {
+                int index = ClientSubscribed(client);
+
+                if (subscribe)
+                {
+                    if (index < 0)
+                    {
+                        _subscribedClients.Add(client);
+                    }
+                }
+                else
+                {
+                    if (index >= 0)
+                    {
+                        _subscribedClients.RemoveAt(index);
+                    }
+                }
+            }
+            SubscribedClientsChanged?.Invoke(this, new object());
+        }
+
+        private GattLocalDescriptor FindDescriptor(UInt16 id)
+        {
+            // Check In built ones first
+            if (_userDescriptionDescriptor != null && _userDescriptionDescriptor._descriptorId == id)
+            {
+                return _userDescriptionDescriptor;
+            }
+
+            // Check PresentationFormats
+            foreach (GattLocalDescriptor desc in _presentationFormatsDescriptors)
+            {
+                if (desc._descriptorId == id)
+                {
+                    return desc;
+                }
+            }
+
+            // Check other descriptors
+            foreach (GattLocalDescriptor desc in _descriptors)
+            {
+                if (desc._descriptorId == id)
+                {
+                    return desc;
+                }
+            }
+            return null;
+        }
+
+        #region Native
+        [MethodImpl(MethodImplOptions.InternalCall)]
+        private extern int NativeNotifyClient(UInt16 connection, UInt16 CharacteristicId, byte[] notifyBuffer);
```

Comment:
catching it here but please replace UIn16 by just ushort
There are few others everywhere. Same for all others, use the normalized version, not the class version

Reply:
Changed all references to UInt16 except for where it relevant to Bluetooth type used. i.e DataWriter. DataReader & PresensationFormatTypes